controllers/users.js

```diff
@@ -0,0 +1,86 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const User = require('../models/users');
+const { JWT_SECRET_KEY } = require('../config');
+const NotFoundError = require('../errors/notFoundError');
+const ConflictError = require('../errors/conflictError');
+
+module.exports.getUsers = (req, res) => {
+  User.find({})
+    .then((users) => res.send({ data: users }))
+    .catch(() => res.status(500).send({ message: 'Произошла ошибка' }));
+};
+
+module.exports.createUser = (req, res) => {
+  const {
+    name, about, avatar, email, password,
+  } = req.body;
+
+  User.countDocuments({ email })
+    .then((count) => {
+      if (count) throw new ConflictError('Пользователь с таким email уже существует в базе');
+
+      bcrypt.hash(password, 10)
+        .then((hash) => User.create({
+          name,
+          about,
+          avatar,
+          email,
+          password: hash,
+        }))
+        .then((user) => res.send({ data: user.omitPrivate() }))
+        .catch((err) => res.status(500).send({ message: `Произошла ошибка при создании пользователя: ${err}` }));
+    })
+    .catch((err) => {
+      const statusCode = err.statusCode || 500;
+      const message = statusCode === 500 ? 'Произошла ошибка' : err.message;
+      res.status(statusCode).send({ message });
+    });
```

Comment:
Можно лучше: использовать единый `.catch()` блок для вложенных промисов.
Особенность работы промисов в том, что `.then()` блоки выполняются в том порядке, в котором они были описаны. Причем вызов следующего `.then()` запускается только после того как промис перейдет в состояние выполнен (_resolved_) или как текущий `.then()` блок будет выполнен. Это означает, что вложенный промис и его `.then()` блоки тоже могут быть встроены в цепочку вызовов, если внутри `.then()` блока будет выполнен `return`.
При наличии одной цепочки вложенный `.catch()` блок можно убирать, потому что любое отклонение промиса (_reject_) будет поймано верхнеуровневым `.catch()` блоком.

```suggestion
      return bcrypt.hash(password, 10)
        .then((hash) => User.create({
          name,
          about,
          avatar,
          email,
          password: hash,
        }))
        .then((user) => res.send({ data: user.omitPrivate() }));
    })
    .catch((err) => {
      const statusCode = err.statusCode || 500;
      const message = statusCode === 500 ? 'Произошла ошибка' : err.message;
      res.status(statusCode).send({ message });
    });
```